pkg/api/licenses.go

```diff
@@ -884,3 +885,306 @@ func SearchInLicense(c *gin.Context) {
+				newLicenseMap["rf_OSIapproved"] = license.OSIapproved.Value
+			}
+			if license.GPLv2compatible.IsDefinedAndNotNull {
+				newLicenseMap["rf_GPLv2compatible"] = license.GPLv2compatible.Value
+			}
+			if license.GPLv3compatible.IsDefinedAndNotNull {
+				newLicenseMap["rf_GPLv3compatible"] = license.GPLv3compatible.Value
+			}
+			if license.Notes.IsDefinedAndNotNull {
+				newLicenseMap["rf_notes"] = license.Notes.Value
+			}
+			if license.Fedora.IsDefinedAndNotNull {
+				newLicenseMap["rf_Fedora"] = license.Fedora.Value
+			}
+			if license.DetectorType.IsDefinedAndNotNull {
+				newLicenseMap["rf_detector_type"] = license.DetectorType.Value
+			}
+			if license.Flag.IsDefinedAndNotNull {
+				newLicenseMap["rf_flag"] = license.Flag.Value
+			}
+			if license.Marydone.IsDefinedAndNotNull {
+				newLicenseMap["marydone"] = license.Marydone.Value
+			}
+			newLicenseMap["external_ref"] = license.ExternalRef
+
+			errMessage, importStatus, newLicense, oldLicense := InsertOrUpdateLicenseOnImport(tx, newLicenseMap)
+
+			if importStatus == models.IMPORT_FAILED {
+				res.Data = append(res.Data, models.LicenseError{
+					Status:    http.StatusInternalServerError,
+					Message:   errMessage,
+					Error:     newLicense.Shortname,
+					Path:      c.Request.URL.Path,
+					Timestamp: time.Now().Format(time.RFC3339),
+				})
+				return errors.New(errMessage)
+			} else if importStatus == models.IMPORT_LICENSE_CREATED {
+				res.Data = append(res.Data, models.LicenseImportStatus{
+					Data:   models.LicenseId{Id: oldLicense.Id, Shortname: oldLicense.Shortname},
+					Status: http.StatusCreated,
+				})
+			} else if importStatus == models.IMPORT_LICENSE_UPDATED {
+				if err := addChangelogsForLicenseUpdate(tx, username, newLicense, oldLicense); err != nil {
+					res.Data = append(res.Data, models.LicenseError{
+						Status:    http.StatusInternalServerError,
+						Message:   "Failed to update license",
+						Error:     newLicense.Shortname,
+						Path:      c.Request.URL.Path,
+						Timestamp: time.Now().Format(time.RFC3339),
+					})
+					return err
+				}
+				res.Data = append(res.Data, models.LicenseImportStatus{
+					Data:   models.LicenseId{Id: newLicense.Id, Shortname: newLicense.Shortname},
+					Status: http.StatusOK,
+				})
+			} else if importStatus == models.IMPORT_LICENSE_UPDATED_EXCEPT_TEXT {
+				if err := addChangelogsForLicenseUpdate(tx, username, newLicense, oldLicense); err != nil {
+					res.Data = append(res.Data, models.LicenseError{
+						Status:    http.StatusInternalServerError,
+						Message:   "Failed to update license",
+						Error:     newLicense.Shortname,
+						Path:      c.Request.URL.Path,
+						Timestamp: time.Now().Format(time.RFC3339),
+					})
+					return err
+				}
+
+				res.Data = append(res.Data, models.LicenseError{
+					Status:    http.StatusConflict,
+					Message:   "All fields except Text were updated. Text was updated manually and cannot be overwritten in an import.",
```

Comment:
Can the `errMessage` be reused here as `InsertOrUpdateLicenseOnImport` already returns same text and it is not used anywhere?